sdk/core/Azure.Core/src/Shared/MutableJsonElement.WriteTo.cs

```diff
@@ -109,5 +118,95 @@ private void WriteArray(string path, int highWaterMark, JsonElement element, Utf
 
             writer.WriteEndArray();
         }
+
+        internal void WritePatch(Utf8JsonWriter writer)
+        {
+            if (!_root.Changes.HasChanges)
+            {
+                return;
+            }
+
+            // This version is not efficient; Proof of concept.
+
+            // TODO: rewrite using Span, so much win there
+            IEnumerable<string> changed = _root.Changes.GetChangedProperties();
+
+            string jsonPath = string.Empty;
+
+            MutableJsonElement current = _root.RootElement;
+
+            // TODO: this breaks for arrays
+            writer.WriteStartObject();
+
+            foreach (string path in changed)
+            {
+                string currentPath = string.Empty;
+                string[] segments = path.Split(MutableJsonDocument.ChangeTracker.Delimiter);
```

Comment:
There are some Span<char> "split" APIs in the BCL. They do the same but without allocations. I wonder if you could use them.

Reply:
Also good to know - I was planning to port this to use Span APIs - I know we are creating a lot of strings in this implementation!

Reply:
It looks like these aren't available until .NET 8: https://learn.microsoft.com/en-us/dotnet/api/system.memoryextensions?view=net-8.0

Reply:
I have addressed this without split APIs in the current implementation.